### ISSUE TITLE
Custom API URL?

### ISSUE THREAD

**Issue by atmomomo** [MichielvanBeers/Flow.Launcher.Plugin.ChatGPT#48]
Great plugin, thanks for the hard work!
Could you please support custom API URL?

**Comment by MichielvanBeers** [comment]
Hi @atmomomo,

Thanks! What do you mean exactly with a custom API URL? What URL would you like to use? Since the plugin is specifically built around the OpenAI chat endpoint, it can't be easily replaced with any kind of API unfortunately.

Michiel

**Comment by atmomomo** [comment]
In practice, we can use the Azure OpenAI API to simulate the function of the original OpenAI API. Unlike OpenAI API, Azure API is available for free. This is the main reason why many, including myself, opt for Azure.

To emulate the same call method as the OpenAI API, we can employ a serverless service, such as a Cloudflare worker, to "convert" the Azure API to mirror the OpenAI API. If you're interested to know how this can be done, please refer to this link: https://atlassc.net/2023/05/30/convert-azure-openai-service-api-to-openai-service-api

In summary, the only modification needed for the launcher ChatGPT plugin is to provide an option for users to input their own custom API URL.

Thank you once again!

**Comment by MichielvanBeers** [comment]
That looks very interesting indeed! I currently am a bit tight on time to implement such a solution, but I welcome a PR to build, test and implement this.

**Comment by MichielvanBeers** [comment]
Closing this issue since it has been marked as stall. Please create a new issue if this is still a problem. Thank you for your contribution!
